sklearn/utils/_repr_html/tests/test_features.py

```diff
@@ -0,0 +1,172 @@
+    ct = ColumnTransformer([("trans", Trans(), [0, 1])])
+    ct.fit(X)
+    estimator_html_repr(ct)
+
+
+def test_n_features_with_MinimalTransformer():
+    X, y = np.array([[0, 1], [1, 1]]), np.array([[0, 1]])
+    ct = ColumnTransformer([("minimal", MinimalTransformer(), [0, 1])])
+    model = Pipeline([("transformer", ct)])
+    model.fit(X, y)
+
+    estimator_html_repr(model)
+
+
+def test_remainder_passthrough():
+    X = np.array([[0, 1], [1, 1]])
+    y = np.array([[0, 1]])
+
+    ct = ColumnTransformer(
+        [("minimal", MinimalTransformer(), [0])], remainder="passthrough"
+    )
+    model = Pipeline([("transformer", ct)])
+    model.fit(X, y)
+    assert "passthrough" in estimator_html_repr(model)
+
+
+def test_features_html_no_fitted_class():
+    """Test that features HTML is generated without fitted CSS class."""
+    features = ["feature1", "feature2", "feature3"]
+    html = _features_html(features)
+
+    assert "3 features" in html
+    assert "feature1" in html
+    assert "feature2" in html
+    assert "feature3" in html
+    assert 'class="features "' in html
+    assert "fitted" not in html
+
+
+def test_features_html_with_fitted_class():
+    """Test that features HTML includes fitted CSS class when specified."""
+    features = ["feature1", "feature2"]
+    html = _features_html(features, is_fitted_css_class="fitted")
+
+    assert "2 features" in html
+    assert 'class="features fitted"' in html
+
+
+def test_features_html_correct_feature_names():
+    """Test that feature names are correctly escaped and displayed in table."""
+    features = ["age", "income", "<script>alert('xss')</script>"]
+    html = _features_html(features)
+
+    assert "age" in html
+    assert "income" in html
+
+    assert "&lt;script&gt;" in html
+    assert "<script>alert" not in html
+    assert "3 features" in html
+
+
+def test_features_html_column_transformer_feature_count():
+    """Test feature count with ColumnTransformer in Pipeline."""
+
+    ct = ColumnTransformer(
+        [
+            ("scaler", StandardScaler(), [0, 1]),
+            ("pca", PCA(n_components=2, random_state=42), [2, 3, 4]),
+        ]
+    )
+
+    X = rng.randn(10, 5)
+    ct.fit(X)
+
+    feature_names = ct.get_feature_names_out()
+    html = _features_html(feature_names)
+
+    assert f"{len(feature_names)} features" in html
+    assert len(feature_names) == 4
+
+
+def test_features_html_with_minimal_transformer():
+    """Test that _features_html works with MinimalTransformer (non-full API)."""
+
+    pipe = Pipeline([("minimal", MinimalTransformer()), ("scaler", StandardScaler())])
+
+    X = rng.randn(10, 3)
+    pipe.fit(X)
+
+    try:
+        feature_names = pipe.get_feature_names_out()
+        html = _features_html(feature_names)
+        assert f"{len(feature_names)} features" in html
+    except AttributeError:
+        features = [f"x{i}" for i in range(3)]
+        html = _features_html(features)
+        assert "3 features" in html
```

Comment:
I find this test weird. Either we do a unittest for `_features_html` and then we do not care about the estimator object generating the feature list, or we do an integration test by inspecting the output of  `estimator_html_repr(pipe)`.

Reply:
Modified.